quartz/scheduler.go

```diff
@@ -94,17 +94,17 @@ func NewStdScheduler() Scheduler {
 func NewStdSchedulerWithOptions(opts StdSchedulerOptions) *StdScheduler {
 	return &StdScheduler{
 		queue:     &priorityQueue{},
+		wg:        &sync.WaitGroup{},
 		interrupt: make(chan struct{}, 1),
 		cancel:    func() {},
 		feeder:    make(chan *item),
-		signal:    make(chan struct{}),
 		dispatch:  make(chan *item),
 		opts:      opts,
 	}
 }
 
 // ScheduleJob schedules a Job using a specified Trigger.
-func (sched *StdScheduler) ScheduleJob(job Job, trigger Trigger) error {
+func (sched *StdScheduler) ScheduleJob(ctx context.Context, job Job, trigger Trigger) error {
```

Comment:
No need for context in the ScheduleJob as the feeder send is not a blocker. Consider removing the cancel and signal fields from the constructor, as they are initialized in the Start method.

Reply:
The signal field is gone and indeed isn't used, but the cancel field is still needed to make sure that the `Stop` method can cancel the go routines that start starts. I have removed the cancel field from the constructor. 

I am a little worried about dropping the context from scheduled job because the feeder channel isn't buffered, and we do a blocking read (and an op that takes a lock) in that thread. This makes it easy to imagine an issue either during shutdown when the feedReader returns (or, somewhat inconceivably panics) if ScheduleJob calls are made from multiple threads. I'm willing to remove the context from the API: it certainly wouldn't be a regression, but it still seems safer.

Reply:
Please add the context parameter to the readme documentation.

Reply:
Done! Thanks for the review and patience!